### ISSUE TITLE
Desktop app crashes upon launch - Mac OS X (10.9.5 - Mavericks)

### ISSUE THREAD

**Issue by zxul767** [nitroshare/nitroshare-desktop#175]
I installed this application simply by using the `nitroshare-0.3.4-osx.dmg` file from the official website.

I investigated a bit and found the crash file with (hopefully) all technical details that may help pin down the problem. Let me know if there's additional information that I can provide.

Thanks

[nitroshare_2018-02-28-142215_Willebaldo-Gomez-Jimenezs-MacBook-Pro-2.crash.txt](https://github.com/nitroshare/nitroshare-desktop/files/1768743/nitroshare_2018-02-28-142215_Willebaldo-Gomez-Jimenezs-MacBook-Pro-2.crash.txt)

**Comment by boomt1337** [comment]
Can you give us your Mac Model?

**Comment by zxul767** [comment]
Sure, it's an old MacBook Pro (13-inch, Mid 2009) running Mavericks:

```
  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro5,5
  Processor Name:	Intel Core 2 Duo
  Processor Speed:	2.53 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache:	3 MB
  Memory:	8 GB
  Bus Speed:	1.07 GHz
  Boot ROM Version:	MBP55.00AC.B03
  SMC Version (system):	1.47f2
  Serial Number (system):	W89330J066J
  Hardware UUID:	82F23266-840E-53D3-80F1-B60B706866F8
  Sudden Motion Sensor:
  State:	Enabled
```

**Comment by boomt1337** [comment]
Alright will see if I can identifiy the issue

**Comment by boomt1337** [comment]
Hmm, seems like an bug. Error code 0x02000148 does happen on iWork as well, but it's not applicable to NitroShare. Will investigate further

**Comment by boomt1337** [comment]
Just for testing purposes, would you please install an later version of NitroShare and try running it? If that does not work, can you try installing the newest NitroShare?

**Comment by zxul767** [comment]
Is there a later version released for Mac OS X? I thought `0.3.4` was the latest release.

**Comment by boomt1337** [comment]
Oof I meant the oldest if possible
		
		

		Get Outlook for iOS
	




On Fri, Mar 9, 2018 at 9:03 PM +1300, "Willebaldo Gómez" <notifications@github.com> wrote:












Is there a later version released for Mac OS X? I thought 0.3.4 was the latest release.

**Comment by boomt1337** [comment]
Oof I meant older
		
		

		Get Outlook for iOS
	




On Fri, Mar 9, 2018 at 9:03 PM +1300, "Willebaldo Gómez" <notifications@github.com> wrote:












Is there a later version released for Mac OS X? I thought 0.3.4 was the latest release.

**Comment by zxul767** [comment]
I just tried version `0.3.3` and it does work. It loads correctly and I can make transfers. They seem a bit slow, but that's another issue.

**Comment by boomt1337** [comment]
Weird. Is this now an version-specific crash?

**Comment by boomt1337** [comment]
I recommend you to use `0.3.3` for now until an patch for this arrives. Seems like 0.3.4 doesn't like Mavericks ATM.

**Comment by zxul767** [comment]
Yes, I started using that version in the meantime. Thanks!

**Comment by boomt1337** [comment]
@nathan-osman this might be an compatability issue. Can you take a look?

**Comment by Go-rom** [comment]
Hello

I'm running Mac OS 10.9 too (macbook pro mid-2010).
I can start nitroshare 0.3.4 and send files (i've run a test with a pdf file from my macbook to my pc under linux mint 18)

But Nitroshare crashes when i try do display the settings.

Here's the crash report

> Process:      nitroshare [11640]
> Path:            /Applications/nitroshare.app/Contents/MacOS/nitroshare
> Identifier:      com.NathanOsman.NitroShare
> Version:         0
> Code Type:       X86-64 (Native)
> Parent Process:  launchd [240]
> Responsible:     nitroshare [11640]
> User ID:         501
> 
> Date/Time:       2018-03-18 12:58:23.494 +0100
> OS Version:      Mac OS X 10.9.5 (13F1911)
> Report Version:  11
> Anonymous UUID:  11A8EC10-B401-C85E-1EEF-00E0A308BA50
> 
> Sleep/Wake UUID: 9C3B46BD-60FB-4695-B705-565C5324B4BF
> 
> Crashed Thread:  0  Dispatch queue: com.apple.main-thread
> 
> Exception Type:  EXC_BAD_INSTRUCTION (SIGILL)
> Exception Codes: 0x0000000000000001, 0x0000000000000000
> 
> Application Specific Information:
> objc[11640]: objc_removeExceptionHandler() called with unknown alt handler; this is probably a bug in multithreaded AppKit use. Set environment variable OBJC_DEBUG_ALT_HANDLERS=YES or break in objc_alt_handler_error() to debug.

I tried 0.3.3 and it's allright, no crash.

BTW, thanks for this useful software !

**Comment by boomt1337** [comment]
@Go-rom again, compatability issues. Still weird, in a way

**Comment by nathan-osman** [comment]
Sorry for the delay. It seems like the problem is that 0.3.4 was compiled against Qt 5.9, which does not support OS X 10.9. NitroShare can be compiled with an older version of Qt so I will try to do that soon and update the DMG on the website. I'll update this issue once that is complete.

**Comment by nathan-osman** [comment]
Alright, I recompiled 0.3.4 with Qt 5.8 and uploaded it. You should be able to download the file from the website and use it on OS X 10.9 now. Please let me know if you run into any further issues.
